src/nemos/basis/_basis_mixin.py

```diff
@@ -13,6 +17,166 @@
 from ..convolve import create_convolutional_predictor
 from ._transformer_basis import TransformerBasis
 
+if TYPE_CHECKING:
+    from ._basis import Basis
+
+
+def set_input_shape_state(method):
```

Comment:
why not make this more general and accept a list of attributes to copy, which default to `["_n_basis_input_", "_input_shape_"]`?